packages/sync/src/modules/Terms.php

```diff
@@ -275,4 +285,22 @@ public function expand_term_taxonomy_id( $args ) {
 			'previous_end' => $previous_end,
 		);
 	}
+
+	/**
+	 * Appends a term object to a row object from the term_relationships table.
+	 * This aids WordPress.com in re-syncing term_relationship tables that are out of sync.
+	 *
+	 * @access public
+	 *
+	 * @param object $relationship A row object from the term_relationships table.
+	 * @return object A row object from the term_relationships table with corresponding term object appended.
+	 */
+	public function add_term_to_relationship( $relationship ) {
+		return (object) array(
+			'object_id'        => $relationship->object_id,
+			'term_taxonomy_id' => $relationship->term_taxonomy_id,
+			'term_order'       => $relationship->term_order,
+			'term'             => get_term_by( 'term_taxonomy_id', $relationship->term_taxonomy_id ),
```

Comment:
Should we have any sort of error handling for `get_term_by` here? Just in the edge case when the relationship is there, but the term is not (for some reason).

Reply:
I'm not sure what's best to do in this situation.
For now, i've added some improved docblocks, as wells as some basic error handling server side.